tests/rptest/tests/consumer_group_balancing_test.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.kafka_cli_consumer import KafkaCliConsumer
+from rptest.services.rpk_producer import RpkProducer
+from rptest.tests.redpanda_test import RedpandaTest
+from ducktape.utils.util import wait_until
+from xxhash import xxh64
+import jump  # package jump-consistent-hash
+from math import ceil, floor
+
+cgroup_balancing_test_num_brokers = 3
+cgroup_balancing_test_num_groups = 3
+
+
+class ConsumerGroupBalancingTest(RedpandaTest):
+    def __init__(self, test_ctx, *args, **kwargs):
+        self._ctx = test_ctx
+        self.producer = None
+        super(ConsumerGroupBalancingTest, self).__init__(
+            test_ctx,
+            num_brokers=cgroup_balancing_test_num_brokers,
+            *args,
+            extra_rp_conf={
+                # disable leader balancer until even distribution control
+                # of __consumer_offsets leaders is implemented
+                "enable_leader_balancer": False,
+                # only partition distribution is tested yet, so we need
+                # a sigle replica per __consumer_offsets partition.
+                # this will change with implementation of
+                # even distribution control of __consumer_offsets leaders
```

Comment:
Even without leader balancing, we'd still expect the r=3 case to achieve uniform balancing across nodes, right?

Seems like this test should use a >3 cluster node count and r=3 to validate the realistic case

Reply:
> Even without leader balancing, we'd still expect the r=3 case to achieve uniform balancing across nodes, right?

As per my understanding, uniform CGC balancing w/o uniform leader balancing is only guaranteed when `(# of __consumer_groups partitions) * r <= (# of nodes)`. So yes, it's possible for `r=3` case but with much bigger number of nodes -- 9 nodes for RP brokers in case of this test. I thought that's too much to ask for CI, but if not then it makes sense to do so.